lib/database/mongodb.ts

```diff
@@ -29,15 +30,23 @@ export default (mPool: MongoClient): MongoDbService => {
     getWord: (wordId: ObjectId) =>
       getWordsCollection().findOne({ _id: new ObjectId(wordId) }),
 
-    getWords: async ({ page, itemsPerPage }: PaginatedWordsRequest) => {
-      const getPaginatedPromise = getWordsCollection()
-        .find()
+    getWords: async ({ page, itemsPerPage, tags }: PaginatedWordsRequest) => {
+      const getTagsFilter = () => {
+        if (tags === undefined) return {};
+        else if (tags.length === 0)
+          return { $or: [{ tags: undefined }, { tags: [] }] };
+        else return { tags: { $all: [...tags] } };
```

Comment:
Is it the same as `return { tags: { $all: tags } }`? I don't think you need to spread unless there is some TS business going on there or you want to make sure you are creating a new array.

Reply:
There was some TypeScript business going on indeed, I can double-check, but I had to spread to make it type-check.

Reply:
Yep, it does not typecheck - we can take a look together, but I didn't dig too deep into this.